modules/afprog/afprog.c

```diff
@@ -106,12 +135,22 @@ afprogram_popen(AFProgramProcessInfo *process_info, GIOCondition cond, gint *fd)
       return FALSE;
     }
 
+  if (pipe(sync_pipe) == -1)
+    {
+      msg_error("Error creating program pipe",
+                evt_tag_str("cmdline", process_info->cmdline->str),
+                evt_tag_error(EVT_TAG_OSERROR));
+      return FALSE;
```

Comment:
Shouldn't we close `msg_pipe` in case of failure?

Reply:
yes, we should.